### PR TITLE
Foundation: homogenise importing C, add MSVCRT

### PR DESCRIPTION
**Pull request by compnerd**
This is preliminary setup for enabling the Windows port.  Ensure that we
import MSVCRT in all the locations where we are importing the C library.
Take the opportunity to homogenise the import for linux/android/cygwin
as well.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: Foundation/AffineTransform.swift [hunk 1]
```diff
@@ -9,8 +9,10 @@
 
 #if os(macOS) || os(iOS)
```
Comment (jakepetroules) [inline Foundation/AffineTransform.swift]: Why are tvOS and watchOS missing here...?

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Well, that's part of the problem.  You can compile glibc for a number of platforms, and just because you can import the module does not mean it is correct.  As an example: I may be building for Linux, but importing glibc might be incorrect because I could have glibc and musl available and wish to build against musl.

**Comment by jakepetroules** [comment]
OK, but `#if os(Linux)` doesn't solve that problem either, since it still hardcodes Glibc. I still think you should generally use `canImport` and later if we want to enable alternates we can do so via additional compiler flags.

**Comment by compnerd** [comment]
There was a single instance of `canImport`, and everywhere else was just spelt out, so I think that switching to `canImport` can be a separate change/conversation.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @phausler @karwa @spevans

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by spevans** [comment]
Looks good on macOS in Xcode

**Comment by compnerd** [comment]
Well, the thing is that nothing really prevents me from building msvcrt on Darwin and then having the msvcrt module available on Darwin.  Why should Darwin be given precedence then?  Why not try to load msvcrt (even on macOS) and use that?  The module can be imported!

**Comment by spevans** [comment]
Having the libc module with different names on different platforms does seems a potential source of problems. It a shame that an alias cant be created and shipped with the toolchain so that a simple `import Libc` could be used everywhere.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**